### PR TITLE
Prevent Stats Indexer reporting error if repo dir missing

### LINKED ISSUE #18847: Error on dexer/stats/queue.go
### Gitea Version

1.15.11

### Git Version

2.7.4

### Operating System

ubuntu/xenial

### How are you running Gitea?

i run gitea by systemd /etc/systemd/system/gitea.service

```systemd
[Unit]
Description=Gitea (Git with a cup of tea)
After=syslog.target
After=network.target
After=mysqld.service

[Service]

LimitMEMLOCK=infinity
LimitNOFILE=65535
RestartSec=2s
Type=simple
User=git
Group=git
WorkingDirectory=/home/git/gitea
ExecStart=/home/git/gitea/gitea web
Restart=always
Environment=USER=git HOME=/home/git

[Install]
WantedBy=multi-user.target
```

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Description

Hello
I see this error in the gitea log
What can I do to correct it?

```
2022/02/21 15:05:06 ...dexer/stats/queue.go:24:handle() [E] stats queue indexer.Index(409) failed: no such file or directory
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by zeripath**
Repositories missing their directory should not report an error from the stats
indexer.

Close #18847

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by Gusted — APPROVED** [review]

**Comment by zeripath** [comment]
Could be backported to 1.16 - but should not hold up releases

File: modules/indexer/stats/db.go [hunk 1]
```diff
@@ -37,6 +37,9 @@ func (db *DBIndexer) Index(id int64) error {
 
 	gitRepo, err := git.OpenRepositoryCtx(ctx, repo.RepoPath())
 	if err != nil {
+		if err.Error() == "no such file or directory" {
+			return nil
+		}
```
Comment (silverwind) [inline modules/indexer/stats/db.go:42]: Can we not compare against `os.ErrNotExist`?

Comment (jolheiser) [reply modules/indexer/stats/db.go:42]: I think we are comparing against https://github.com/zeripath/gitea/blob/d3bf78ba3c67c5ed7beceb886a3f2540dc816c4e/modules/git/repo_base_nogogit.go#L50

That being said, comparing to error strings seems brittle. Can we instead export a concrete error to check against?

Comment (silverwind) [reply modules/indexer/stats/db.go:42]: I think the error string is incorrect too, as I think the meaning is there is no repo directory at the given path. I think we should create a custom error for this and compare using `errors.Is`.

Comment (zeripath) [reply modules/indexer/stats/db.go:42]: This error string is chosen to match the error string returned by go-git

Comment (silverwind) [reply modules/indexer/stats/db.go:42]: Ok, in that case our hands are bound I guess.

**Review by silverwind — APPROVED** [review]

**Pull request merged**